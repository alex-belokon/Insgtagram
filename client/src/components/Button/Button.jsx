import React, {memo} from 'react'

const Button = ({ onClick, type, children }) => {
    return (
        <button 
        data-testid="button-attr" 
        className={`button  ${type ? 'button--' + type : ''}`} 
        onClick={onClick} >
            {children}
        </button>
    )
}

export default memo(Button)
