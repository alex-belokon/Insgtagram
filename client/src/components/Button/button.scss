.button {
  @import "~bootstrap/scss/bootstrap";
}
