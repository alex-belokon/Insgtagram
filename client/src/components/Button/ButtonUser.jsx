import React from "react";
import "./button.scss";
import { Button, Row } from "react-bootstrap";

const ButtonUser = ({ text, className }) => {
  return (
    <div className="button">
      <Row className="mx-0">
        <Button data-testid="button" type="button" className={className}>
          {text}
        </Button>
      </Row>
    </div>
  );
};

export default ButtonUser;
