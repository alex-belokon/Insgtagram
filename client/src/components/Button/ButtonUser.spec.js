import Button from './ButtonUser.jsx';
import { render} from '@testing-library/react'



describe('Button component', () => {
 
    it('should render correctly', () => {
        const { getByRole } = render(<Button />)
        getByRole('button')
    })
    it('snapshot', () => {
        const { getByRole } = render(<Button>Decrement</Button>);
        expect(getByRole('button')).toMatchSnapshot()
    })
    
    it('should add class with type', () => {
            const type = 'primary';
            const { queryByRole } = render(<Button type={type}>Increment</Button>)
            const button = queryByRole('button')
            expect(button.className).toEqual(expect.stringContaining('btn btn-primary'))
        })
 
})