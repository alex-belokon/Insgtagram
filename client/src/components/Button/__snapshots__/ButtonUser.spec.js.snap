// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component snapshot 1`] = `
<button
  class="btn btn-primary"
  data-testid="button"
  type="button"
/>
`;
